### PR TITLE
fixed some happy little accidents

### PR DESCRIPTION
**Pull request by sjka**
Signed-off-by: Simon Kaufmann <simon.kfm@googlemail.com>

### CODE REVIEW

File: bundles/automation/org.eclipse.smarthome.automation.module.script.rulesupport/src/main/java/org/eclipse/smarthome/automation/module/script/rulesupport/internal/delegates/SimpleConditionHandlerDelegate.java [hunk 1]
```diff
@@ -22,11 +22,11 @@
  */
 public class SimpleConditionHandlerDelegate extends BaseModuleHandler<Condition> implements ConditionHandler {
 
-    private org.eclipse.smarthome.automation.module.script.rulesupport.shared.simple.SimpleConditionHandler conditionHandler;
+    private SimpleConditionHandler conditionHandler;
 
     public SimpleConditionHandlerDelegate(Condition condition, SimpleConditionHandler scriptedHandler) {
         super(condition);
-
+        this.conditionHandler = scriptedHandler;
```
Comment (maggu2810) [inline bundles/automation/org.eclipse.smarthome.automation.module.script.rulesupport/src/main/java/org/eclipse/smarthome/automation/module/script/rulesupport/internal/delegates/SimpleConditionHandlerDelegate.java]: Using `final` on the declaration would help to identify such errors.

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/i18n/ThingStatusInfoI18nLocalizationService.java [hunk 2]
```diff
@@ -87,7 +87,7 @@ protected void setTranslationProvider(TranslationProvider i18nProvider) {
     }
 
     protected void unsetTranslationProvider(TranslationProvider i18nProvider) {
-        i18nProvider = null;
+        this.i18nProvider = null;
```
Comment (maggu2810) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/i18n/ThingStatusInfoI18nLocalizationService.java]: Using `final` on the method argument would help to identify such errors.

**Comment by sjka** [comment]
> Using final on the method argument would help to identify such errors.

let's split out this discussion -> #3651

**Comment by maggu2810** [comment]
> let's split out this discussion -> #3651

Sure, I already approved it (regardless if `final` is used or not), only wait for Travis CI.
Thanks for opening a discussion.

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/i18n/ThingStatusInfoI18nLocalizationService.java (hunk 2 above)
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/i18n/ThingStatusInfoI18nLocalizationService.java]: @maggu2810 Just by chance I noticed on that line that the search&replace from I18nProvider->TranslationProvider was only a partial success: It is a bit nasty that all variables are still called i18nProvider :-(

Comment (maggu2810) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/i18n/ThingStatusInfoI18nLocalizationService.java]: You are right, the variable names has been not replaced and so they are not really correct.
But this does not introduce that bug at all.

Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/i18n/ThingStatusInfoI18nLocalizationService.java]: No, definitely unrelated to the bug, that's not what I wanted to convey!

**Pull request merged**